### PR TITLE
wip on usage command

### PR DESCRIPTION
**Pull request by uovobw**
this code does _not_ compile at the moment.
My plan is to be able to create all the commands and the command dispatcher _outside_ of the loop in the main thread and use their reference as a command to be able to address them in the command dispatcher. i am not sure this is the right way to go, i fear it might require us to rework the whole command interface. any comments are welcome.

the change in itself is not huge, since rust does not support variadic functions and iterating on a Vec of Commands cannot be done (or i haven't been able to) the plan was to get the usage at the beginning of the main() and then return that each time it's needed. 
missing:
- privmsg handling
- destination type routing in both telegram and irc tranports
- decent logging

### CODE REVIEW

**Comment by crisidev** [comment]
I'll take a look to it later or maybe tomorrow. 
If I get it right the idea is just to reuse the commands and the dispatcher in the main loop without instantiating them every time.. I  think it is the right way to approach the problem. About the implementation, this is a playground and for me rewriting the full command interface is something we will have to do to allow mote reusable code!

File: holysee/src/commands/command_dispatcher.rs [hunk 1]
```diff
@@ -7,30 +7,34 @@ use message::Message;
...
 
 pub struct CommandDispatcher<'a> {
-    command: Box<Command + 'a>,
+    command: &'a Command,
     enabled_commands: &'a Vec<String>,
 }
 
 impl<'a> CommandDispatcher<'a> {
-    pub fn new(settings: &'a settings::Commands) -> CommandDispatcher<'a> {
+    pub fn new(settings: &'a settings::Commands, base_command: &'a Command) -> CommandDispatcher<'a> {
```
Comment (crisidev) [inline holysee/src/commands/command_dispatcher.rs]: Do we really need to pass the `base_command` to the constructor?

File: holysee/src/commands/usage.rs [hunk 2]
```diff
@@ -0,0 +1,84 @@
+extern crate regex;
+
+use chan::Sender;
+use std::collections::HashMap;
+
+use self::regex::Regex;
+
+use message::{Message, TransportType,DestinationType};
```
Comment (crisidev) [inline holysee/src/commands/usage.rs]: Missing space between `TransportType` and `DestinationType`

**Review by crisidev — APPROVED** [review]
This looks good to me.. I like having commands initialisation inside outside of the loop.

**Comment by uovobw** [comment]
some more work in progress. abstracted the name and moved some functions in the `Command` trait.
there is a bug now introduced with the refactor: the if/else/if/else... chain in the main is gone, so now all commands are relayed no matted what, we need to find a way of marking the filter (like the last seen or the relay commands) so that it runs differently than the normal command

**Comment by crisidev** [comment]
LGTM. I think we can merge this!

**Pull request merged**